Apps/Src/Fans.c

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
+
+/* Fans.c
+Controls speed of fans which cool down the battery pack.
+Uses Pins PC6,7 and PB14,15
+*/
+
+#include "Fans.h"
+#include "Contactor.h"
+
+void Fans_Init(void){
+    #ifdef SIMULATION
+		Simulator_Log(LOG_INFO, "Fans Initialized\n");
+	#endif
+    BSP_PWM_Init();
+}
+
+/**
+ * @brief This function will change the speed of the fans
+ * @param fan Number of fan to change speed (1-4)
+ * @param speed Speed of Fan(0-8)
+ * @return ErrorStatus will return 1 if successful, 0 if error occurred
+ */
+
+ErrorStatus Fans_Set(uint8_t fan, uint32_t speed){
+    //don't mess with the contactor
+    if (fan == CFAN) return ERROR;
+    //Range of pulse is 0-4000
+    //First check to make sure that change is within range of values
+    //Load new value into Compare and Capture Register
+    return (speed < 0 || speed > TOPSPEED) ? ERROR : BSP_PWM_Set(fan, speed * DIVIDER); //if not in range, leave function with error code
```

Comment:
I agree. However, I do want to point out that ideally, we would error-handle all function calls. But this would essentially mean entering the fault state and setting all fans to max speed anyways, so we might as well do it here.